client/app/components/QueryEditor.jsx

```diff
@@ -203,6 +224,13 @@ class QueryEditor extends React.Component {
   };
 
   updateQuery = (queryText) => {
+    if (this.updateQueryTimeout) {
+      clearTimeout(this.updateQueryTimeout);
+    }
+    const doneCallback = this.props.updateQueryDone;
+    this.updateQueryTimeout = setTimeout(() => {
+      doneCallback(queryText);
+    }, 3000);
```

Comment:
this is where you set the timeout for running dry run if I understand correctly - could that be more clear in a way and also possibly extract the value itself to somewhere which is more config like ? 
or at least give this value a const of it's own.

Reply:
No problem, I'll make it a const